Tooltip preference.

digi0ps opened the pull request:
Enable/Disable tooltip from Popup and Settings.

@oliversauter Have used the already available `common-ui` toggle switch. Is it fine?

![image](https://user-images.githubusercontent.com/3889675/40368796-343b6714-5dfa-11e8-9ddc-e718e2eb4f91.png)

![image](https://user-images.githubusercontent.com/3889675/40368849-475c734c-5dfa-11e8-8ff5-165a77729e5d.png)

oliversauter commented:
Looks really neat! :) Great and fast work. 

Can we maybe have a text inside the switch saying something like "on" and "off"
Also, for UX reasons, can we, as soon as a user has changed the setting, close the popup?
Maybe with a 500ms delay? This way users can immediately continue working on the document. From a UX perspective I don't see a reason why you would first enable the Memex.Link features, and then continue doing stuff on the popup.

oliversauter commented:
Ah and also it would be good to have a small info icon behind "tooltip" so that we can link to one of the knowledge base articles. 

Link i put there once we launch, just prepare everything and tell me in here, which line I should look for the url. Thanks :) 

[noun_1228105_3EB995.svg.zip](https://github.com/WorldBrain/Memex/files/2027198/noun_1228105_3EB995.svg.zip)

digi0ps commented:
- Added "on"/"off" text inside toggle
- Added info icon with the url inside `src/content-tooltip/constants.js`
- Close popup when disabled

![image](https://user-images.githubusercontent.com/3889675/40506064-e0e31a60-5fb3-11e8-9425-efc1f25b1f9f.png)

![image](https://user-images.githubusercontent.com/3889675/40506070-eab5ac6a-5fb3-11e8-84dc-e1767366b40e.png)

oliversauter commented:
Looks great, but some things still need some work :)

Would do some minor alignment adjustments:
1) make the info symbol vertically aligned with the text
2) make the on/off horizontally aligned with the space between the round white button and the edge of the pill. 
3) I see that you implemented a timer for automatic closing of the popup when the turning off of the tooltip, i would also add one for turning it on. 
4) Turning off doesn't really work. I can still make highlights and it shows me the tooltip after turning off. 
5) where can I change the link for the info button?

Really important to be aware to test through all relevant user interactions before making a PR, and that stuff works. And if you make a PR with knowing that things don't work yet, then note it down in the PR. Otherwise people reviewing take time to describe an issue you are already aware of and it costs everyone unnecessary time to do all that much more time. 
No problem really though, in case that sounds rough here in written form. Just give the functionalities that are affected by your work always a proper check before doing a PR in the future. Something really important for a professional web development workflow. 
Happy hacking. :)

digi0ps commented:
Turning it off works perfectly for me. Were you getting any errors in your console? I will double check to make sure it's working. Sorry about this! 

And you can change the link for the info button in `src/content-tooltip/constants.js`. I have it stored in a variable named *INFO_URL*. 

I will make the other mentioned changes.
